internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,189 @@
+	Namespace     string
+	Kubeclient    kubernetes.Interface
+
+	grpcPort int32
+}
+
+// Create returns a bundle registry pod built from an index image
+func (rp *RegistryPod) Create(ctx context.Context) (*corev1.Pod, error) {
+	// Make the pod spec
+	pod, err := rp.build()
+	if err != nil {
+		return nil, fmt.Errorf("error in building registry pod: %v", err)
+	}
+
+	// Check if Pod already exists
+	tmpPod, err := rp.Kubeclient.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Create pod in kubernetes cluster using the above pod definition and clientset
+			pod, err = rp.Kubeclient.CoreV1().Pods(pod.Namespace).Create(ctx, pod, metav1.CreateOptions{})
+			if err != nil {
+				return nil, fmt.Errorf("error creating registry pod: %v", err)
+			}
+		} else {
+			return nil, fmt.Errorf("error getting existing registry pod: %v", err)
+		}
+	} else {
+		return tmpPod, nil
+	}
+
+	// Upon creation of new pod, poll and verify that pod status is running
+	podCheck := wait.ConditionFunc(func() (done bool, err error) {
+		p, err := rp.Kubeclient.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return true, fmt.Errorf("error getting pod %s: %w", pod.Name, err)
+		}
+		return p.Status.Phase == corev1.PodRunning, nil
+	})
+
+	err = wait.PollImmediateUntil(time.Duration(1*time.Second), podCheck, ctx.Done())
+	if err != nil {
+		return nil, fmt.Errorf("error waiting for registry pod %s to run: %v", pod.Name, err)
+	}
+
+	//TODO: make pod logs available, maybe return log output to caller
+	_, err = rp.fetchPodLogs(ctx, pod)
+	if err != nil {
+		return pod, fmt.Errorf("error in fetching pod logs: %v", err)
+	}
+
+	return pod, err
+}
+
+func (rp *RegistryPod) setDefaults() {
+	// set the grpcPort to default 50051
+	rp.grpcPort = defaultGRPCPort
+
+	if len(rp.IndexImage) == 0 {
+		rp.IndexImage = defaultIndexImage
+	}
+
+	if len(rp.BundleAddMode) == 0 {
+		if rp.IndexImage == defaultIndexImage {
+			rp.BundleAddMode = "semver"
+		} else {
+			rp.BundleAddMode = "replaces"
+		}
+	}
+}
+
+func (rp *RegistryPod) validate() error {
+	if len(rp.BundleImage) == 0 {
+		return fmt.Errorf("bundle image is required")
+	}
+	if len(rp.DBPath) == 0 {
+		return fmt.Errorf("registry database path is required")
+	}
+
+	if len(rp.Namespace) == 0 {
+		return fmt.Errorf("pod namespace is required")
+	}
+
+	return nil
+}
+
+func makePodName(bundleName string) (string, error) {
+	var podName string
+	if strings.Contains(bundleName, "/") {
+		split := strings.Split(bundleName, "/")
+		lastSegment := strings.Split(split[len(split)-1:][0], ":")
+		// get the last but one element from last segment excluding the version number and append index
+		podName = lastSegment[len(lastSegment)-2] + "-index"
+	} else {
+		return "", fmt.Errorf("invalid bundle image name: %s", bundleName)
+	}
+	return podName, nil
+}
+func (rp *RegistryPod) build() (*corev1.Pod, error) {
```

Comment:
```suggestion
func (rp *RegistryPod) newCoreV1RegistryPod() (*corev1.Pod, error) {
```

Reply:
from the previous review feedback, renamed it to `build` as it is more appropriate since it is a method on RegistryPod struct.

Reply:
The builds does not clarify what it is doing and returning and I cannot understand the argumentation `it is more appropriate since it is a method on RegistryPod struct.` I disagree, however, it is a nit. So, 👍